src/ansi.rs

```diff
@@ -78,6 +78,23 @@ fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
     }
 }
 
+fn parse_number(input: &[u8]) -> Option<u8> {
+    if input.is_empty() {
+        return None;
+    }
+    let mut index: u8 = 0;
```

Comment:
Might want to rename index to `num` or something similar since this is a general number parsing method.